db/models/articles.model.js

```diff
@@ -82,3 +82,18 @@ exports.changeVotes = (article_id, inc_votes) => {
       })
   }
 
+exports.removeComment = (comment_id) => {
+    return db.query(
+        `SELECT * FROM comments
+        WHERE comment_id = $1`, 
+        [comment_id])
+        .then(({ rows }) => {
+            if (!rows.length) {
+                return Promise.reject({ status: 404, msg: "Not Found" })
+            } else 
+        return db.query (
+            `DELETE FROM comments 
+            WHERE comment_id = $1`, 
+            [comment_id])
+        })
```

Comment:
I can see why you've done the above - but you could do this with one query by using RETURNING on a DELETE . It should return the deleted rows - then you can check whether they're empty.